transaction/src/main/java/org/wildfly/httpclient/transaction/HttpRemoteTransactionService.java

```diff
@@ -261,12 +267,33 @@ protected void handleImpl(HttpServerExchange exchange, ImportResult<LocalTransac
         }
     }
 
+    private void internalSendException(HttpServerExchange exchange, int status, Throwable e) {
+        try {
+            exchange.setStatusCode(status);
+            exchange.getResponseHeaders().put(Headers.CONTENT_TYPE, EXCEPTION.toString());
+
+            final Marshaller marshaller = httpServiceConfig.getHttpMarshallerFactory(exchange).createMarshaller();
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            final ByteOutput byteOutput = new NoFlushByteOutput(Marshalling.createByteOutput(outputStream));
+            // start the marshaller
+            marshaller.start(byteOutput);
+            marshaller.writeObject(e);
+            marshaller.write(0);
+            marshaller.finish();
+            marshaller.flush();
+            exchange.getResponseSender().send(ByteBuffer.wrap(outputStream.toByteArray()));
+        } catch (IOException e1) {
+            HttpRemoteTransactionMessages.MESSAGES.debugf(e, "Failed to write exception");
+        }
+    }
+
+    @Deprecated
```

Comment:
deprecated instead of deleted just because this class is public api in WlidFly (hence the intenralSendException above, it could be just sendException...)